backend/src/scripts/utility/teamspaces/removeTeamspaceAndOwner.js

```diff
@@ -56,37 +33,47 @@ const removeUserFromAllTeamspaces = async (user) => {
 };
 
 const removeUser = async (user) => {
-	await removeUserFromAllTeamspaces(user);
-	await Promise.all([
-		dropCollection('loginRecords', user),
-		dropCollection('notifications', user),
-		dropUser(user),
-	]);
+	const userExists = await getUserByUsername(user).catch(() => false);
+
+	if (userExists) {
+		await removeUserFromAllTeamspaces(user);
+		await deleteUser(user);
+	}
 };
 
-const run = async (users) => {
-	const userArr = users.split(',');
-	for (const user of userArr) {
-		logger.logInfo(`-${user}`);
-		// eslint-disable-next-line no-await-in-loop
-		await removeTeamspace(user);
-		// eslint-disable-next-line no-await-in-loop
-		await removeUser(user);
+const run = async (teamspaces, removeOwners) => {
+	const teamspaceArr = teamspaces.split(',');
+	const tsList = await getTeamspaceList();
+	for (const teamspace of teamspaceArr) {
+		logger.logInfo(`-${teamspace}`);
+		if (tsList.includes(teamspace)) {
+			// eslint-disable-next-line no-await-in-loop
+			await removeTeamspace(teamspace);
+		}
+		if (removeOwners) {
```

Comment:
Hmmm I don't agree here 😆 

If we can't find the teamspace, the owner might still exist. This can happen if we removed the teamspace by running the script once with removeOwner set to false, then run it again to remove the owner account.